Adjust account-specific terraform to manually-applied changes.

adunkman opened the pull request:
These tweaks have been made to update our logging infrastructure; apply them to terraform to bring things back in line. This change goes hand-in-hand with an update to our [Court-specific environment documentation](https://github.com/ustaxcourt/ef-cms/wiki/AWS-environment-configuration) to document these values.

adunkman committed:
Adjust account-specific terraform to reality.

These tweaks have been made to update our logging infrastrucutre; apply
them to terraform to bring things back in line.

adunkman committed:
Add migration lambda logs to kibana.

This one’s for you, @mmarcotte!

mmarcotte reviewed (approved):
![super](https://media.giphy.com/media/zL6Xl2A5R8yNG/giphy.gif)

Thanks for this Andrew!

adunkman committed:
Ignore changes to cognito_identity_providers.

AWS Elasticsearch requires that it sets up its own identity provider and
user pool client.

adunkman commented:
Added one more commit to fix #496: 

- When deploying a domain, AWS Elasticsearch creates a Cognito user pool client and its mapped identity provider, configured to point at the domain endpoint. 
- This creates an issue when Terraform runs against the Cognito identity pool, since it sees no such mapped identity provider and removes it. (Reported at #496). This breaks authentication for Kibana when the run happens. 
- Any time the domain is deployed — on changes to its volumes, instances, configuration, etc — it re-creates the user pool client and mapped identity provider if it doesn’t exist. This fixes authentication. 

…

FIGHT! 

- I initially attempted to bring these resources under Terraform control, which would allow us to manage them ourselves as we do all other AWS resources. This creates a cycle:
  - To configure the user pool client, you need the endpoint for Elasticsearch.
  - To determine the endpoint for Elasticsearch, you need the user pool client.
- My attempted fix was to use a known domain for the Elasticsearch endpoint (`logs.ef-cms.ustaxcourt.gov`) to break the cycle. This worked great! Except…
  - Even when Elasticsearch is configured to use a custom domain (https://github.com/hashicorp/terraform-provider-aws/issues/16059), it still forces the Cognito user pool client and its mapped identity provider to point to the AWS-managed domain. The one that we can’t know until after the Elasticsearch domain is created. 

There is no permanent fix without a change in AWS behavior (https://github.com/hashicorp/terraform-provider-aws/issues/5557) —  https://github.com/ustaxcourt/ef-cms/pull/532/commits/c853d8bcfcb2eeafe15624cd073b411696f88d0d implements a workaround (to have Terraform ignore changes to `cognito_identity_providers`, sidestepping the fight).

adunkman committed:
Fix bash scripting bug preventing log subscription

This prevented adding log subscriptions to all of the court’s
environments.

adunkman commented:
Since this PR is still open, tacked on one more commit to fix a bash scripting bug which we discovered right before our load test yesterday — the IF-statement conditional is reversed for determining if `LOG_GROUP_ENVIRONMENTS` is set, which ensured the default value was always used. Fixed in https://github.com/ustaxcourt/ef-cms/pull/532/commits/de5aa717196d42560e5e5c71e7d1536bf3ea4a8a.
